pandas/io/common.py

```diff
@@ -844,6 +876,116 @@ def get_handle(
     )
 
 
+# error: Definition of "__exit__" in base class "TarFile" is incompatible with
+# definition in base class "BytesIO"  [misc]
+# error: Definition of "__enter__" in base class "TarFile" is incompatible with
+# definition in base class "BytesIO"  [misc]
+# error: Definition of "__enter__" in base class "TarFile" is incompatible with
+# definition in base class "BinaryIO"  [misc]
+# error: Definition of "__enter__" in base class "TarFile" is incompatible with
+# definition in base class "IO"  [misc]
+# error: Definition of "read" in base class "TarFile" is incompatible with
+# definition in base class "BytesIO"  [misc]
+# error: Definition of "read" in base class "TarFile" is incompatible with
+# definition in base class "IO"  [misc]
+class _BytesTarFile(tarfile.TarFile, BytesIO):  # type: ignore[misc]
```

Comment:
umm can we inherit something similar to BytesZip and make this easier (or create a base class)

Reply:
@twoertwein also [commented this](https://github.com/pandas-dev/pandas/pull/44787#discussion_r770164095). I found it hard to implement, as there's not as much shared logic between those classes as it might seem in first glance.